### PR TITLE
fix(object): fix test regression

### PR DESCRIPTION
**Pull request by Monitob**

### CODE REVIEW

File: scaleway/resource_object_bucket_test.go [hunk 1]
```diff
@@ -323,19 +333,22 @@ func TestAccScalewayObjectBucket_Cors_Delete(t *testing.T) {
 }
 
 func TestAccScalewayObjectBucket_Cors_EmptyOrigin(t *testing.T) {
+	if !*UpdateCassettes {
+		t.Skip("Skipping ObjectStorage test as this kind of resource can't be deleted before 24h")
```
Comment (Monitob) [inline scaleway/resource_object_bucket_test.go:337]: @remyleone I'll skip this tests to be used with cassettes for now.

File: scaleway/resource_object_bucket_test.go [hunk 2]
```diff
@@ -391,32 +404,35 @@ func testAccCheckScalewayObjectBucketCors(tt *TestTools, n string, corsRules []*
 func testAccCheckScalewayObjectBucketExists(tt *TestTools, n string) resource.TestCheckFunc {
 	return func(state *terraform.State) error {
 		rs := state.RootModule().Resources[n]
-		bucketName := rs.Primary.Attributes["name"]
+		if rs != nil {
```
Comment (remyleone) [inline scaleway/resource_object_bucket_test.go:407]: Could you have the test of rs == nil return an error and then we avoid imbricating all that code in this conditional?

File: scaleway/resource_object_bucket_test.go [hunk 3]
```diff
@@ -323,19 +333,22 @@ func TestAccScalewayObjectBucket_Cors_Delete(t *testing.T) {
...
+	bucketName := sdkacctest.RandomWithPrefix("test-acc-scaleway-object-bucket-cors-empty-origin")
 	resource.ParallelTest(t, resource.TestCase{
 		PreCheck:          func() { testAccPreCheck(t) },
 		ProviderFactories: tt.ProviderFactories,
 		CheckDestroy:      testAccCheckScalewayObjectBucketDestroy(tt),
 		Steps: []resource.TestStep{
 			{
 				Config: fmt.Sprintf(`
-					resource "scaleway_object_bucket" "bucket" {
+					resource "scaleway_object_bucket" "%s" {
```
Comment (remyleone) [inline scaleway/resource_object_bucket_test.go:351]: Does that mean that all assertions will be managed with string substitution as well? Why is this change needed?

File: scaleway/resource_object_bucket_test.go [hunk 4]
```diff
@@ -302,7 +312,7 @@ func TestAccScalewayObjectBucket_Cors_Delete(t *testing.T) {
...
```
Comment (remyleone) [inline scaleway/resource_object_bucket_test.go:315]: Why is this change needed?

File: scaleway/resource_object_bucket_test.go [hunk 5]
```diff
@@ -165,19 +167,23 @@ func testSweepStorageObjectBucket(_ string) error {
...
+	bucketName := sdkacctest.RandomWithPrefix("test-acc-scaleway-object-bucket-cors-update")
 	resource.ParallelTest(t, resource.TestCase{
 		PreCheck:          func() { testAccPreCheck(t) },
 		ProviderFactories: tt.ProviderFactories,
 		CheckDestroy:      testAccCheckScalewayObjectBucketDestroy(tt),
 		Steps: []resource.TestStep{
 			{
 				Config: fmt.Sprintf(`
-					resource "scaleway_object_bucket" "bucket" {
+					resource "scaleway_object_bucket" "%s" {
```
Comment (remyleone) [inline scaleway/resource_object_bucket_test.go:186]: Why is this change needed?

File: scaleway/resource_object_bucket_test.go [hunk 6]
```diff
@@ -22,30 +23,31 @@ func init() {
...
-					resource "scaleway_object_bucket" "base" {
+					resource "scaleway_object_bucket" "base-01" {
 						name = "%s"
 						tags = {
 							foo = "bar"
 						}
 					}
 
-					resource "scaleway_object_bucket" "ams-bucket" {
+					resource "scaleway_object_bucket" "ams-bucket-01" {
```
Comment (remyleone) [inline scaleway/resource_object_bucket_test.go:50]: Why is this change needed?

File: scaleway/resource_object_bucket_test.go [hunk 7]
```diff
@@ -54,51 +56,51 @@ func TestAccScalewayObjectBucket_Basic(t *testing.T) {
 						}
 					}
 
-					resource "scaleway_object_bucket" "par-bucket" {
+					resource "scaleway_object_bucket" "par-bucket-01" {
```
Comment (remyleone) [inline scaleway/resource_object_bucket_test.go:59]: Why is this change needed?

**Review by remyleone — APPROVED** [review]
